backend/router/stories.py

```diff
@@ -35,7 +36,23 @@ async def create_story(
     db: Session = Depends(get_db),
 ):
     token_data = await main.get_token_if_present(request)
-    db_story = crud.create_story(db=db, story=story, token_data=token_data)
+    story_to_update = None
```

Comment:
It's a little hard to see at first glance what this piece of code does
What do you think of this solution?

```python
token_data = await main.get_token_if_present(request)
user = get_user(db, token_data)
story_to_update = get_existing_story(user, token_data, db)

if story_to_update:
    db_story = crud.update_story(db=db, story_id=story_to_update, story=story)
else:
    db_story = crud.create_story(db=db, story=story, user=user)

```

having the following methods
```python
def get_user(db, token_data):
    if token_data and isinstance(token_data, auth_schemas.UserToken):
        return get_user_by_email(db, email=token_data.email)
    else:
        return None


def get_existing_story(user, token_data, db):
    if user and user.story:
        return user.story.id

    if isinstance(token_data, auth_schemas.StoryToken):
        return token_data.story_id

    return None
```

Even, perhaps we could have this `get_user` method in user module, so it can be re-used in `auth/main.get_current_story`

Reply:
pity we don't have test on this, I could have checked that the refactor does't breaks anything

It would be nice to have test on this! 🙂

Reply:
There, it's neater!

It would, indeed, be lovely to add tests about this. After having done the whole `travels` thing I think I know how to go about it, but I need that merged to avoid creating another mega-branch there